conn_pool.go

```diff
@@ -105,7 +112,24 @@ func (p *ConnPool) Acquire() (c *Conn, err error) {
 		if p.logLevel >= LogLevelWarn {
 			p.logger.Warn("All connections in pool are busy - waiting...")
 		}
+
+		deadlineReached := false
+
+		if p.connAllocTimeout > 0 {
+			timer := time.NewTimer(p.connAllocTimeout)
+			defer timer.Stop()
+
+			select {
+			case <-timer.C:
+				deadlineReached = true
+				p.cond.Signal()
+			}
```

Comment:
When all connections are in use, is Acquire() supposed to immediately return once a connection becomes available? It looks like with this new change, the `select` will simply block until the timer has elapsed, with no chance for Acquire() to ever return before waiting the full `connAllocTimeout`.